src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/TransportCreation/TransportCreationHandler.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.Data.SqlClient.SNI;
+using Microsoft.Data.SqlClientX.Handlers.Connection;
+
+namespace Microsoft.Data.SqlClientX.Handlers.TransportCreation
+{
+    internal class TransportCreationHandler : IHandler<ConnectionHandlerContext>
```

Comment:
Sounds good! I'll keep watching and check again once the entire connection flow's done. The TDS streams establish a common view of the Once the TCP/NP/etc. connection's established, the TDS stream provides a common abstraction

Reply:
Sounds good! I'll keep watching and check again once the entire connection flow's done. I suggested deriving `xTransportCreationHandler` classes because although the TDS streams establish a common view of the TCP/NP/etc. connection once it's established, at this stage I think the transport creation handlers will probably take quite different paths.